addons/sale/models/sale_order.py

```diff
@@ -764,10 +779,13 @@ def _send_order_confirmation_mail(self):
         if self.env.su:
             # sending mail in sudo was meant for it being sent from superuser
             self = self.with_user(SUPERUSER_ID)
-        template_id = self._find_mail_template(force_confirmation_template=True)
-        if template_id:
-            for order in self:
-                order.with_context(force_send=True).message_post_with_template(template_id, composition_mode='comment', email_layout_xmlid="mail.mail_notification_paynow")
+        mail_template = self._find_mail_template(force_confirmation_template=True)
+        if mail_template and len(self) > 0:
```

Comment:
```suggestion
        if mail_template:
```

the self check should be moved in the beginning of the method:
```python
if not self:
    return
```

(and btw `len(self) > 0 == bool(self)`)